src/blueprints/datafile.py

```diff
@@ -62,10 +62,17 @@ def display_name(self) -> str:
         """Display name for a datafile (not necessarily unique)"""
         return self.filename
 
+    @property
+    def filepath(self) -> Path:
+        """The full path to the file"""
+        if self.directory is not None:
+            return Path(self.directory) / self.filename
```

Comment:
```suggestion
            return self.directory / self.filename
```

I don't think the Path constructor call is needed here - at this point we've already checked that `self.directory is not None`, so I think `self.directory` is guaranteed to be a `Path`.

Reply:
yes, it looks working fine